noise2same/network.py

```diff
@@ -636,44 +659,44 @@ def __init__(
             out_channels = int(base_channels * (2 ** (i - 1)))
 
             if upsampling[i - 1] == "conv":
-                upsample = partial(conv_transpose, kernel_size=2,
-                                   stride=2,
-                                   bias=True)
+                upsample = partial(conv_transpose, kernel_size=2, stride=2, bias=True)
                 if unit_type == "dual_pass":
 
-                    in_channels_local, out_channels_local, in_channels_global, out_channels_global = divide_channels(
-                        in_channels,
-                        out_channels,
-                        global_ratio)
+                    twin_unit = TwinSample(
+                        upsample, in_channels, out_channels, global_ratio
+                    )
+                    upsampling_block_main = twin_unit.sample_local
+                    upsampling_block_secondary = twin_unit.sample_global
```

Comment:
Same here: you can store skip connections as tuples and make `upsampling_block` to work with tuples through `forward()` in `TwinSample`